src/sign/index.ts

```diff
@@ -1,14 +1,43 @@
 import { createHmac } from "crypto";
 
-export function sign(secret: string, payload: string | object): string {
+export enum Algorithm {
+  SHA1 = "sha1",
+  SHA256 = "sha256",
+}
+
+type SignOptions = {
+  secret: string;
+  algorithm?: string;
+};
+
+export function sign(
+  options: SignOptions | string,
+  payload: string | object
+): string {
+  let secret;
+  let algorithm;
+
+  if (typeof options === "string") {
+    secret = options;
+    algorithm = Algorithm.SHA1;
+  } else {
+    secret = options.secret;
+    algorithm = options.algorithm ?? Algorithm.SHA1;
+  }
```

Comment:
I see this a bit messy, if you come up with a cleaner way to write this, please let me know :)

Reply:
> I see this a bit messy, if you come up with a cleaner way to write this, please let me know :)

@wolfy1339 TS question to the rescue! Any good idea on how to control this in a less hacky way? 😇